### ISSUE TITLE
Make BLAS_CUTOFF configurable by environment variables.

### ISSUE THREAD

**Issue by thecooldrop** [rust-ndarray/ndarray#883]
Hello everyone, 

here I would like to propose to make the BLAS_CUTOFF for matrix multiplications configurable by environment variables, or in some other way, as currently anything with dimensions under 8 really makes a hit to performance of matrix multiplications.

If this is something that maintainers would like to pursue I would offer to implement it as described in following [Stackoverflow question](https://stackoverflow.com/questions/37526598/how-can-i-override-a-constant-via-a-compiler-option). I would probably pursue the path explained with build script, as that is already there.

Maybe this issue can be used as a chance to make other constants configurable as well if such exist?

**Comment by bluss** [comment]
Compile-time environment variable has no cost (we can use option_env and const fn, no build script please) and it's viable. I would prefer to just increase the threshold if you've seen drawbacks with the current one, that's much simpler. Current cutoff is 7. What's your preference?

What size of matrix multiplication problems do you use? We're not really optimized for small (like size 8).

**Comment by thecooldrop** [comment]
Well in my case I would actually like to turn the cutoff completely off, and yes if you are wondering yes I would like to use BLAS for all computations, and all dimensions.

Currently when I run my algorithm with 8x8 matrices I get a run time of ca 400 microseconds per iter, but when I run it with 7x7 matrices I get run time of ca 670 microseconds per iter, and since I am most likely to be running the algorithm with either 4x4 or 8x8 matrices I would like to have this configurable. 

I would not like to simply change the value here, since I really like the ambitions of matrixmultiply crate.

**Comment by bluss** [comment]
Thanks for clarification.

Note about matrixmultiply - its main use case is when blas is disabled or for strides of arrays that blas can't handle. It doesn't target performance for small matrices either, so the intention is that you use one or the other, not really both.

**Comment by thecooldrop** [comment]
Yes, that is clear, but one is left without a choice if matrices of dimension larger and smaller than BLAS_CUTTOFF are used, and that is what brought me here with this feature request.
